### ISSUE TITLE
Question about yout code

### ISSUE THREAD

**Issue by jbvasconcellos** [KMnP/intentonomy#8]
Hello there,

I have two minor questions about `_get_CAM` and `get_CAM,` in the `Localizationloss` class -- the part that has to be changed according to one's model definition. First, you set `out_b = gcam(X, image_ids="")[1]`, but looking into GradCam I wonder if it shouldn't be just `out_b = gcam(X)` in general; this in fact returns a tensor of shape `[bs, num_classes]` as commented.

Later on, you set `gcam_b = gcam.get(layer)` (and `norm_img` as its normalization), a tensor wish shape `[bs, 1, fmpH, fmpW].` As a few lines below, in `get_CAM`, you set `CAM_masks[:, cls_idx, :, :] = self._get_CAM(model, X, cls_idx).squeeze(1)`, and since the masks' shape depend on the `mask_length` parameter (but not on `fmpH` or `fmpW`, of course), I wonder if also `get_CAM` won't depend on the model adopted. That is, wouldn't this part crack if `fmpH` or `fmpW` and `mask_length` don't match?

**Comment by EvelynMuir** [comment]
I met with the same first question.

**Comment by KMnP** [comment]
Thanks for your questions. 

(1) I included `image_ids=` since the forward method of the model need this input parameters since essentially calling `gcam()` traces back to https://github.com/KMnP/intentonomy/blob/master/loc_loss.py#L206.

(2) thanks for pointing this out. The masks' shapes depends on the model itself: since we used 224x224 input size and used the layer4 of a resnet-50 model, thus the produced mask shape is 7 x 7. To make it more generalizable, `self.mask_length` should be a input variable. If you use some other layers than layer4 or change the input sizes, the resulting mask shape will change.

Hope this answers your questions.
